app/controllers/nsm/steps/disbursement_cost_controller.rb

```diff
@@ -29,7 +32,24 @@ def ensure_disbursement
       end
 
       def additional_permitted_params
-        [:apply_vat]
+        [:apply_vat, :add_another]
+      end
+
+      def new_record?
+        disbursement.disbursement_date.nil? || disbursement.miles.nil?
```

Comment:
This is buggy. For "other" disbursement types, disbursement.miles will always be nil, because for those you just enter the total cost, not the mileage. so this will always return true.

Reply:
That makes sense, but I wasn't sure what else to use to determine if it's a new record or not.

Reply:
You need an `&&`, not an `||`. A new record will have neither a number of miles nor a `total_cost_without_vat`, so if _both_ are nil then you know this is new.

Reply:
Ahh you're right